java/vector/src/test/java/org/apache/arrow/vector/TestLargeVector.java

```diff
@@ -0,0 +1,187 @@
+        }
+      }
+      System.out.println("Successfully read " + vecLength + " values");
+    }
+    System.out.println("Successfully released the large vector.");
+  }
+
+  private static void testLargeDecimalVector() {
+    System.out.println("Testing large decimal vector.");
+
+    final long bufSize = 4 * 1024 * 1024 * 1024L;
+    final int vecLength = (int) (bufSize / DecimalVector.TYPE_WIDTH);
+
+    try (BufferAllocator allocator = new RootAllocator(Long.MAX_VALUE);
+         DecimalVector largeVec = new DecimalVector("vec", allocator, 38, 16)) {
+      largeVec.allocateNew(vecLength);
+
+      System.out.println("Successfully allocated a vector with capacity " + vecLength);
+
+      for (int i = 0; i < vecLength; i++) {
+        largeVec.set(i, 0);
+
+        if ((i + 1) % 10000 == 0) {
+          System.out.println("Successfully written " + (i + 1) + " values");
+        }
+      }
+      System.out.println("Successfully written " + vecLength + " values");
+
+      for (int i = 0; i < vecLength; i++) {
+        ArrowBuf buf = largeVec.get(i);
+        assertEquals(buf.capacity(), DecimalVector.TYPE_WIDTH);
+        assertEquals(0, buf.getLong(0));
+        assertEquals(0, buf.getLong(8));
+
+        if ((i + 1) % 10000 == 0) {
+          System.out.println("Successfully read " + (i + 1) + " values");
+        }
+      }
+      System.out.println("Successfully read " + vecLength + " values");
+    }
+    System.out.println("Successfully released the large vector.");
+  }
+
+  private static void testLargeFixedSizeBinaryVector() {
+    System.out.println("Testing large fixed size binary vector.");
+
+    final long bufSize = 4 * 1024 * 1024 * 1024L;
+    final int typeWidth = 8;
+    final int vecLength = (int) (bufSize / typeWidth);
+
+    try (BufferAllocator allocator = new RootAllocator(Long.MAX_VALUE);
+         FixedSizeBinaryVector largeVec = new FixedSizeBinaryVector("vec", allocator, typeWidth)) {
+      largeVec.allocateNew(vecLength);
+
+      System.out.println("Successfully allocated a vector with capacity " + vecLength);
+
+      byte[] value = new byte[] {'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h'};
+      for (int i = 0; i < vecLength; i++) {
+        largeVec.set(i, value);
+
+        if ((i + 1) % 10000 == 0) {
+          System.out.println("Successfully written " + (i + 1) + " values");
+        }
+      }
+      System.out.println("Successfully written " + vecLength + " values");
+
+      for (int i = 0; i < vecLength; i++) {
+        byte[] buf = largeVec.get(i);
+        assertEquals(typeWidth, buf.length);
+        assertArrayEquals(buf, value);
+
+        if ((i + 1) % 10000 == 0) {
+          System.out.println("Successfully read " + (i + 1) + " values");
+        }
+      }
+      System.out.println("Successfully read " + vecLength + " values");
+    }
+    System.out.println("Successfully released the large vector.");
+  }
+
+  public static void main(String[] args) {
```

Comment:
same here about making these standard unit tests

Reply:
As discussed above, if we make it a unit test, we cannot allocate too much memory.